aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -342,6 +351,73 @@ module aptos_framework::account {
+            abort error::invalid_argument(EINVALID_CAPABILITY_TYPE)
+        };
+    }
+
+    /// offer signer capability to another address
+    public entry fun offer_signer_capability(account: &signer, signer_capability_sig_bytes: vector<u8>, account_scheme: u8, account_public_key_bytes: vector<u8>, recipient_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(recipient_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+        assert!(account_scheme == 0 || account_scheme == 1, error::invalid_argument(EINVALID_SCHEME));
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        // proof that this account intends to delegate its signer capability to another account
+        let signer_capability_offer_proof_challenge = SignerCapabilityOfferProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            recipient_address,
+        };
+
+        // verify that the `SignerCapabilityOfferProofChallenge` is correct and signed by the account owner's private key
+        if (account_scheme == ED25519_SCHEME) {
+            let pubkey = ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+            let matching_auth_key_to_public_key_bytes = ed25519::unvalidated_public_key_to_authentication_key(&pubkey);
+            assert!(account_resource.authentication_key == matching_auth_key_to_public_key_bytes, error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY));
+
+            let signer_capability_sig = ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+            assert!(ed25519::signature_verify_strict_t(&signer_capability_sig, &pubkey, signer_capability_offer_proof_challenge), error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+        } else {
+            let pubkey = multi_ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+            let matching_auth_key_to_public_key_bytes = multi_ed25519::unvalidated_public_key_to_authentication_key(&pubkey);
+            assert!(account_resource.authentication_key == matching_auth_key_to_public_key_bytes, error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY));
+
+            let signer_capability_sig = multi_ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+            assert!(multi_ed25519::signature_verify_strict_t(&signer_capability_sig, &pubkey, signer_capability_offer_proof_challenge), error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+        };
+
+        // update the existing signer capability offer or put in a new signer capability offer for the current account
+        update_or_add_capability_offer<SignerCapability>(addr, recipient_address);
+    }
+
+    /// Accept signer capability from `offerer_address` if there's an existing signer capability offer to the account owner
+    public fun accept_signer_capability(account: &signer, offerer_address: address) : signer acquires Account {
```

Comment:
create_authorized_signer -- this is where @areshand's comments can be better leveraged.

Reply:
okie just updated! thanks so much for reviewing : )